### DISCUSSION TITLE
How to add coin to vector and simple map

### DISCUSSION THREAD

**Discussion by Drizzle210** [aptos-labs/aptos-developer-discussions#291] [Questions]
### Describe your question in detail.

I am implementing a function like this that uses simple map to map tokens and vector to store tokens. But I can't do that at the same time because the token has been moved and token can't copy. How to fix this, I'm new to move, thank you very much
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/55441313/5c5b6662-a6a2-4bb1-8eed-03e1ca08ab2e)
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/55441313/e8f3b536-9f0d-4cc5-a4a0-b2072f136edf)

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment]
Your SimpleMap has an `address` key (this should be `Coin<CoinType>`, right?). You're inserting a `Coin<CoinType>` in `simple_map::add` where the key parameter is. 

Also you can just add the `copy` ability to the `Record` struct.
